### PR TITLE
Relations

### PR DESCRIPTION
**Pull request by submarcos**
Principle : 
- set relations manual / intersection or distance between layers
- compute queryset on the fly
- store results in FeatureRelation
- manually sync relations, or use celery taskworker to automatic run with async

### CODE REVIEW

File: geostore/migrations/0034_auto_20191209_0902.py [hunk 1]
```diff
@@ -1,22 +0,0 @@
-# Generated by Django 2.2.7 on 2019-12-09 09:02
```
Comment (LePetitTim) [inline geostore/migrations/0034_auto_20191209_0902.py]: Problem with the migration 0034_auto_20191209_0902.py, it should not be squash with layerrelations migrations.

File: geostore/migrations/0035_layerextrageom_editable.py [hunk 2]
```diff
@@ -1,18 +0,0 @@
-# Generated by Django 2.2.8 on 2019-12-10 10:45
```
Comment (LePetitTim) [inline geostore/migrations/0035_layerextrageom_editable.py]: This migration should not be remove. The migration 0034_auto_20200107_1240.py should be after this one without the 2 operations about layerextrageom

File: doc/use.rst [hunk 3]
```diff
@@ -47,13 +47,14 @@ Without validation
   from django.contrib.geos.geometries import GEOSGeometry
 
   layer = Layer.objects.create(name='Mushroom spot 2')
-  feature = Feature.objects.create(layer=layer,
-                                   geom=GEOSGeometry("POINT(0 0)")
+  feature = Feature(layer=layer,
+                    geom=GEOSGeometry("POINT(0 0)")
+  feature.clean()  # ok
```
Comment (LePetitTim) [inline doc/use.rst:52]: - Add information about modelform, viewsets that clean then save.
- Add information that you need to savec after the clean

File: geostore/tests/test_models/test_relations.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                             [])
+
+    def test_bad_relation(self):
+        bad_relation = LayerRelation.objects.create(
+            relation_type='distance',
+            origin=self.layer_city,
+            destination=self.layer_trek,
+            settings={'distance': 1000}
+        )
+        self.city_cover.sync_relations(bad_relation.pk)
```
Comment (LePetitTim) [inline geostore/tests/test_models/test_relations.py:66]: Add comment, relation is city to trek not the opposite

File: doc/use.rst (hunk 3 above)
Comment (submarcos) [reply doc/use.rst:52]: done

**Pull request merged**